handbook/engineering/code-insights/index.md

```diff
@@ -0,0 +1,71 @@
+
+**Value to Sourcegraph:** The Sourcegraph sales cycle is unusual because although we consistently wow our users, the economic decision-maker is usually not one of these users. Instead, the people with the power to sign a contract with us are higher up within an organization and usually depend on running a trial to fully understand the value of Sourcegraph. When code insights answers these personas' higher-level questions about codebases that our core features do not currently answer, it can dramatically speed up our sales cycle as well as our sales pipeline by giving decision makers a reason to buy Sourcegraph without needing to run a trial.
+
+**Milestones and Outcomes:**
+
+_Because code insights is an entirely new feature and closely informed by customer feedback, the further in the future goals get, the more flexible the order of these milestones is (especially when it comes to interleaving "business" milestones like beta/GA/paid, with feature milestones, like third-party data or monitoring/campaign integration)._ 
+
+1. ✅ Three customers give us qualitative feedback after using our code insights prototypes to guide the initial product direction. 
+
+    - **Outcome:** We have a list of potential features and their likely value that we can use to achieve our initial adoption milestones ([Productboard view](https://sourcegraph.productboard.com/feature-board/2327586-code-insights-next-objective)). 
+
+1. 🔄 We have decided on and implemented metrics to quantitatively measure the adoption of code insights prototypes (see [RFC 279](https://docs.google.com/document/d/1I10tm5CFZvzQYNeV--JacRGLLIUesXQBp6ZO8uhakRs/edit#)). 
+
+    - **Outcome:** We have weekly quantitative reports on the use of code insights at each customer. 
+
+1. Code insights can scale to large (if not the largest) codebases. 
+
+    - **Outcome:** All but the largest customers can use insights prototypes over their entire codebase to answer questions about the state of _all_ of their code.
+
+1. Code insights provides the functionality and positioning to be immediately useful to many customers (this may include integrations with campaigns, code monitoring, or third party data sources – see [Productboard feature view](https://sourcegraph.productboard.com/feature-board/1793095-code-insights)).  
+
+    - **Outcome:** Code insights features are actively in use by VPs/Directors at 5+ enterprise customers.  
+
+1. Customers can easily create their own custom insights in an intentionally-designed UI.
```

Comment:
> in an intentionally-designed UI.

What does this mean?

Reply:
I should have said "dedicated UI" for clarity, even if the this milestone's UI form won't be as "dedicated" as something like the campaigns UI yet.

Reply:
So like a WYSIWYG editor for creating code insights?

Reply:
Sort of, though I don't know what WYSIWYG means when WYS is code and WYG is a visual graph (is Looker/Bigquery what you mean?) because it might still be code-as-UI here. The core of this is "you don't have to go and set it up like a prototype in your settings file" but "there's a UI that we built for this purpose" even if this isn't its final form as we learn about problems or personas who would prefer a different UI. 

The simplest first step would be to just pull out the code insights settings code into its own settings-like editor you access somewhere in the UI so you don't get distracted or confused by a busy interface when setting up an insight.